### ISSUE TITLE
CDT 9.10.0 Endgame

### ISSUE THREAD

**Issue by jonahgraham** [eclipse-cdt/cdt-infra#23]
This is the Release plan and TODO list for CDT 9.10.0 release, part of the Eclipse 2019-12 Simrel. CDT makes its simrel contribution on +1 day in the release cycle (normally Monday by 10pm UTC).

## Steps for M1

Items before M1 +1 day:
- [x] Create release on [PMI](https://projects.eclipse.org/projects/tools.cdt)
- [x] Create [New and Noteworthy page](https://wiki.eclipse.org/CDT/User/NewInCDT) for the release
- [x] Update cdt.target to point to new dependency versions
- [x] Synchronize CDT.setup to cdt.target
- [x] Update API baseline to last release and resolve any API errors
- [ ] Remove old API error filters if they are no longer relevant
- [x] Synchronize CDT.setup for API baseline
- [x] Update `help-docs-eclipserun-repo` in root pom.xml and bump versions of all the docs plug-ins
- [x] Increment version of all feature.xml, pom.xml and any other place full version is used. (Easiest way is global find and replace, e.g. `s/9.9.0/9.10.0/g` and review changes.)
- [x] Increment version URL in releng/org.eclipse.cdt-feature/p2.inf
- [x] Increment version URL in EPP's packages/org.eclipse.epp.package.cpp.feature/p2.inf
- [x] Update `comparator.repo` in root pom.xml to last release and if there have been any changes since the branch, bump versions of MANIFEST.MFs
- [x] Update Maven versions (check [CI job](https://ci.eclipse.org/cdt/view/Releng/job/cdt-check-maven-plugin-updates/))
- [x] Update `simrel-site` in root pom.xml
- [x] Ensure the build is stable - it is always better to release a "Green Dot" build
- [x] Ensure all previous [Endgame issues](https://github.com/eclipse-cdt/cdt-infra/labels/endgame) are done.

Items on M1 +0 day:
- [x] If tracking most recent platform, look for announcement and update cdt.target to M1 Eclipse Platform

Items on M1 +1 day:
- [x] [Promote a cdt build from jenkins](https://ci.eclipse.org/cdt/view/Releng/job/promote-a-build/)
    - if cdt.target has been updated to Platform M1, then STANDALONE can be checked, otherwise do it on +4
- [x] Smoke test the release by installing it into platform's SDK build.
- [x] Add description to the promote-a-build job and the job it promoted.
- [x] Mark promoted job as Keep forever
- [x] Using the [CBI aggregator tooling](https://wiki.eclipse.org/CBI/aggregator) update [cdt.aggrcon](https://git.eclipse.org/c/simrel/org.eclipse.simrel.build.git/tree/cdt.aggrcon). An example of a previous contribution is in [gerrit](https://git.eclipse.org/r/#/c/135069/)
- [x] Notify the cdt-dev list that the build was posted. [Example on cdt-dev archives](https://www.eclipse.org/lists/cdt-dev/msg33395.html)

Items on M1 +4 day (or shortly after):
- [x] The final part of the release is to wait for announcement (normally on +4 day) like [this](https://www.eclipse.org/lists/epp-dev/msg05332.html) on [epp-dev](https://accounts.eclipse.org/mailing-list/epp-dev) and test the EPP build and confirm operation with a +1 to epp-dev mailing 
- [x] If standalone was not included on +1 day [Promote a cdt build from jenkins](https://ci.eclipse.org/cdt/view/Releng/job/promote-a-build/) to add the standalone build (STANDALONE_ONLY = true)
- [x] Add description to the promote-a-build job and the job it promoted.
- [x] Mark promoted job as Keep forever
- [x] Reply to email sent on +1 day with links to EPP and standalone debugger. [Example on cdt-dev archives](https://www.eclipse.org/lists/cdt-dev/msg33872.html).

## Steps for M2

Items on M2 +0 day:
- [x] If tracking most recent platform, look for announcement and update cdt.target to M2 Eclipse Platform

Items on M2 +1 day:
- [x] [Promote a cdt build from jenkins](https://ci.eclipse.org/cdt/view/Releng/job/promote-a-build/)
    - if cdt.target has been updated to Platform M2, then STANDALONE can be checked, otherwise do it on +4
- [x] Smoke test the release by installing it into platform's SDK build.
- [x] Add description to the promote-a-build job and the job it promoted.
- [x] Mark promoted job as Keep forever
- [x] Using the [CBI aggregator tooling](https://wiki.eclipse.org/CBI/aggregator) update [cdt.aggrcon](https://git.eclipse.org/c/simrel/org.eclipse.simrel.build.git/tree/cdt.aggrcon). An example of a previous contribution is in [gerrit](https://git.eclipse.org/r/#/c/135069/)
- [x] Notify the cdt-dev list that the build was posted. [Example on cdt-dev archives](https://www.eclipse.org/lists/cdt-dev/msg33395.html)

Items on M2 +4 day (or shortly after):
- [x] The final part of the release is to wait for announcement (normally on +4 day) like [this](https://www.eclipse.org/lists/epp-dev/msg05332.html) on [epp-dev](https://accounts.eclipse.org/mailing-list/epp-dev) and test the EPP build and confirm operation with a +1 to epp-dev mailing 
- [x] If standalone was not included on +1 day [Promote a cdt build from jenkins](https://ci.eclipse.org/cdt/view/Releng/job/promote-a-build/) to add the standalone build (STANDALONE_ONLY = true)
- [x] Add description to the promote-a-build job and the job it promoted.
- [x] Mark promoted job as Keep forever
- [x] Reply to email sent on +1 day with links to EPP and standalone debugger. [Example on cdt-dev archives](https://www.eclipse.org/lists/cdt-dev/msg33872.html).

## Steps for M3

Items before M3 +1 day:
- [x] Ensure release entry on [PMI](https://projects.eclipse.org/projects/tools.cdt) "Release Date" section it says the appropriate "This release is part of Eclipse IDE ??????". 
- [x] Schedule a release review if it is needed (generally >= 12 months since the last one). Consult [EDP](https://www.eclipse.org/projects/dev_process/) for guidance.

Items on M3 +0 day:
- [x] If tracking most recent platform, look for announcement and update cdt.target to M3 Eclipse Platform

Items on M3 +1 day:
- [x] [Promote a cdt build from jenkins](https://ci.eclipse.org/cdt/view/Releng/job/promote-a-build/)
    - if cdt.target has been updated to Platform M3, then STANDALONE can be checked, otherwise do it on +4
- [x] Smoke test the release by installing it into platform's SDK build.
- [x] Add description to the promote-a-build job and the job it promoted.
- [x] Mark promoted job as Keep forever
- [x] Using the [CBI aggregator tooling](https://wiki.eclipse.org/CBI/aggregator) update [cdt.aggrcon](https://git.eclipse.org/c/simrel/org.eclipse.simrel.build.git/tree/cdt.aggrcon). An example of a previous contribution is in [gerrit](https://git.eclipse.org/r/#/c/135069/)
- [x] Notify the cdt-dev list that the build was posted. [Example on cdt-dev archives](https://www.eclipse.org/lists/cdt-dev/msg33395.html)

Items on M3 +4 day (or shortly after):
- [x] The final part of the release is to wait for announcement (normally on +4 day) like [this](https://www.eclipse.org/lists/epp-dev/msg05332.html) on [epp-dev](https://accounts.eclipse.org/mailing-list/epp-dev) and test the EPP build and confirm operation with a +1 to epp-dev mailing 
- [x] If standalone was not included on +1 day [Promote a cdt build from jenkins](https://ci.eclipse.org/cdt/view/Releng/job/promote-a-build/) to add the standalone build (STANDALONE_ONLY = true)
- [x] Add description to the promote-a-build job and the job it promoted.
- [x] Mark promoted job as Keep forever
- [x] Reply to email sent on +1 day with links to EPP and standalone debugger. [Example on cdt-dev archives](https://www.eclipse.org/lists/cdt-dev/msg33872.html).

## Steps for RC1

Items before RC1 +1 day:
- [x] Ensure there are no API errors

Items on RC1 +0 day:
- [x] If tracking most recent platform, look for announcement and update cdt.target to RC1 Eclipse Platform

Items on RC1 +1 day:
- [x] [Promote a cdt build from jenkins](https://ci.eclipse.org/cdt/view/Releng/job/promote-a-build/)
    - if cdt.target has been updated to Platform RC1, then STANDALONE can be checked, otherwise do it on +4
- [x] Smoke test the release by installing it into platform's SDK build.
- [x] Add description to the promote-a-build job and the job it promoted.
- [x] Mark promoted job as Keep forever
- [x] Using the [CBI aggregator tooling](https://wiki.eclipse.org/CBI/aggregator) update [cdt.aggrcon](https://git.eclipse.org/c/simrel/org.eclipse.simrel.build.git/tree/cdt.aggrcon). An example of a previous contribution is in [gerrit](https://git.eclipse.org/r/#/c/135069/)
- [x] Notify the cdt-dev list that the build was posted. [Example on cdt-dev archives](https://www.eclipse.org/lists/cdt-dev/msg33395.html)

Items on RC1 +4 day (or shortly after):
- [x] The final part of the release is to wait for announcement (normally on +4 day) like [this](https://www.eclipse.org/lists/epp-dev/msg05332.html) on [epp-dev](https://accounts.eclipse.org/mailing-list/epp-dev) and test the EPP build and confirm operation with a +1 to epp-dev mailing 
- [x] If standalone was not included on +1 day [Promote a cdt build from jenkins](https://ci.eclipse.org/cdt/view/Releng/job/promote-a-build/) to add the standalone build (STANDALONE_ONLY = true)
- [x] Add description to the promote-a-build job and the job it promoted.
- [x] Mark promoted job as Keep forever
- [x] Reply to email sent on +1 day with links to EPP and standalone debugger. [Example on cdt-dev archives](https://www.eclipse.org/lists/cdt-dev/msg33872.html).

## Steps for RC2

These steps should be done before RC2 release, they can be completed at anytime between the last branch creation/release and RC2.

Items before RC2 +1 day:
- [x] Create branch for CDT release
- [x] Create new [Jenkinsfiles](https://github.com/eclipse-cdt/cdt-infra/tree/master/jenkins/pipelines/cdt) for new release and create jobs
- [x] Make sure [New and Noteworthy page](https://wiki.eclipse.org/CDT/User/NewInCDT) is up to date

Items on RC2 +0 day:
- [x] If tracking most recent platform, look for announcement and update cdt.target to RC2 Eclipse Platform

Items on RC2 +1 day:
- [x] [Promote a cdt build from jenkins](https://ci.eclipse.org/cdt/view/Releng/job/promote-a-build/)
    - if cdt.target has been updated to Platform RC2, then STANDALONE can be checked, otherwise do it on +4
- [x] Smoke test the release by installing it into platform's SDK build.
- [x] Add description to the promote-a-build job and the job it promoted.
- [x] Mark promoted job as Keep forever
- [x] Using the [CBI aggregator tooling](https://wiki.eclipse.org/CBI/aggregator) update [cdt.aggrcon](https://git.eclipse.org/c/simrel/org.eclipse.simrel.build.git/tree/cdt.aggrcon). An example of a previous contribution is in [gerrit](https://git.eclipse.org/r/#/c/135069/)
- [x] Notify the cdt-dev list that the build was posted. [Example on cdt-dev archives](https://www.eclipse.org/lists/cdt-dev/msg33395.html)

Items on RC2 +4 day (or shortly after):
- [x] The final part of the release is to wait for announcement (normally on +4 day) like [this](https://www.eclipse.org/lists/epp-dev/msg05332.html) on [epp-dev](https://accounts.eclipse.org/mailing-list/epp-dev) and test the EPP build and confirm operation with a +1 to epp-dev mailing 
- [x] If standalone was not included on +1 day [Promote a cdt build from jenkins](https://ci.eclipse.org/cdt/view/Releng/job/promote-a-build/) to add the standalone build (STANDALONE_ONLY = true)
- [x] Add description to the promote-a-build job and the job it promoted.
- [x] Mark promoted job as Keep forever
- [x] Reply to email sent on +1 day with links to EPP and standalone debugger. [Example on cdt-dev archives](https://www.eclipse.org/lists/cdt-dev/msg33872.html).

## Steps for Release

Items for quiet week (between RC2 and release):
- [x] Make sure documentation is part of simrel's [help](http://help.eclipse.org).

Items to be done 2 days before release:
- [x] [Promote a cdt build from jenkins](https://ci.eclipse.org/cdt/view/Releng/job/promote-a-build/) to releases (run twice so standalone is promoted too if needed)
- [x] Add description to the promote-a-build job and the job it promoted.
- [x] Unmark as keep all old Milestone and RC jobs
- [x] Update or create [composites](https://github.com/eclipse-cdt/cdt-infra/tree/master/download/releases) in cdt-infra in preparation for going public on release day

Release day:
- [x] Verify we get a GO on the cross-project-issues-dev mailing list
- [x] [Promote the files to download](https://ci.eclipse.org/cdt/view/Releng/job/promote-files-to-download/) - this makes the release public
- [x] Tag the release. Example: `git tag -a CDT_9_4_2 2cdc63eae52c8952c0d2543e1e31ca6e25225c4a -m"CDT 9.4.2" && git push origin CDT_9_4_2`
- [x] Update https://www.eclipse.org/cdt/ (current status) and https://www.eclipse.org/cdt/downloads.php
- [x] Make an announcement on cdt-dev. [Example on cdt-dev archives](https://www.eclipse.org/lists/cdt-dev/msg33618.html)

Items after the release:
- [x] Delete download.eclipse.org/tools/cdt/builds for this release (e.g:
```
# Keep a backup on archive temporarily
ssh build.eclipse.org rsync -aAXv /home/data/httpd/download.eclipse.org/tools/cdt/builds /home/data/httpd/archive.eclipse.org/tools/cdt
ssh build.eclipse.org mv -v /home/data/httpd/archive.eclipse.org/tools/cdt/builds/9.10 /home/data/httpd/archive.eclipse.org/tools/cdt/builds/9.10-scheduled-for-deletion
ssh build.eclipse.org rm -rv /home/data/httpd/download.eclipse.org/tools/cdt/builds/9.10 
```
- [x] Using the [CBI aggregator tooling](https://wiki.eclipse.org/CBI/aggregator) update [cdt.aggrcon](https://git.eclipse.org/c/simrel/org.eclipse.simrel.build.git/tree/cdt.aggrcon) to point to the release as it is not probably pointing to the cdt/builds directory. (e.g. https://git.eclipse.org/r/#/c/154853/)

**Comment by jonahgraham** [comment]
> Remove old API error filters if they are no longer relevant

This wasn't done for 9.10, it is done for 9.11 (on master) already.

**Comment by jonahgraham** [comment]
That's it. CDT 9.10.0 is done.
